src/theory/arith/arith_msum.h

```diff
@@ -0,0 +1,185 @@
+   *
+   * This function returns false if the m-variable of n is already
+   * present in n.
+   */
+  static bool getMonomial(Node n, std::map<Node, Node>& msum);
+
+  /** get monomial sum for real-valued term n
+   *
+   * If this function returns true, it sets msum to a monmoial sum such that
+   *   [msum] is equivalent to n
+   *
+   * This function may return false if n is not a sum of monomials
+   * whose variables are pairwise unique.
+   * If term n is in rewritten form, this function should always return true.
+   */
+  static bool getMonomialSum(Node n, std::map<Node, Node>& msum);
+
+  /** get monmoial sum literal for literal lit
+   *
+   * If this function returns true, it sets msum to a monmoial sum such that
+   *   [msum] <k> 0  is equivalent to lit[0] <k> lit[1]
+   * where k is the Kind of lit, one of { EQUAL, GEQ }.
+   *
+   * This function may return false if either side of lit is not a sum
+   * of monomials whose variables are pairwise unique on that side.
+   * If literal lit is in rewritten form, this function should always return
+   * true.
+   */
+  static bool getMonomialSumLit(Node lit, std::map<Node, Node>& msum);
+
+  /** make node for monomial sum
+   *
+   * Make the Node corresponding to the interpretation of msum, [msum], where:
+   *   [msum] = sum_{( v, c ) \in msum } [c]*[v]
+   */
+  static Node mkNode(const std::map<Node, Node>& msum);
+
+  /** make coefficent term
+   *
+   * Input coeff is a m-constant.
+   * Returns the term t if coeff.isNull() or coeff*t otherwise.
+   */
+  static Node mkCoeffTerm(Node coeff, Node t);
+
+  /** isolate variable v in constraint ([msum] <k> 0)
+   *
+   * If this function returns a value ret where ret != 0, then
+   * veq_c is set to m-constant, and val is set to a term such that:
+   *    If ret=1, then ([veq_c] * v <k> val) is equivalent to [msum] <k> 0.
+   *   If ret=-1, then (val <k> [veq_c] * v) is equivalent to [msum] <k> 0.
+   *   If veq_c is non-null, then it is a positive constant Rational.
+   * The returned value of veq_c is only non-null if v has integer type.
+   *
+   * This function returns 0 indicating a failure if msum does not contain
+   * a (non-zero) monomial having mvariable v.
+   */
+  static int isolate(
+      Node v, const std::map<Node, Node>& msum, Node& veq_c, Node& val, Kind k);
+
+  /** isolate variable v in constraint ([msum] <k> 0)
+   *
+   * If this function returns a value ret where ret != 0, then veq
+   * is set to a literal that is equivalent to ([msum] <k> 0), and:
+   *    If ret=1, then veq is of the form ( v <k> val) if veq_c.isNull(),
+   *                   or ([veq_c] * v <k> val) if !veq_c.isNull().
+   *   If ret=-1, then veq is of the form ( val <k> v) if veq_c.isNull(),
+   *                   or (val <k> [veq_c] * v) if !veq_c.isNull().
+   * If doCoeff = false or v does not have Integer type, then veq_c is null.
+   *
+   * This function returns 0 indiciating a failure if msum does not contain
```

Comment:
indicating (-i)

Reply:
What do you mean here?

Reply:
Your wrote indiciating instead of indicating.

Reply:
Ah I was interpreting this as negation of quantity i (too much thinking about arithmetic)